### ISSUE TITLE
Warn for misplaced `# ruff: noqa: ` directives

### ISSUE THREAD

**Issue by aaron-skydio** [astral-sh/ruff#17726]
### Summary

Related to these, but distinct I think:
- https://github.com/astral-sh/ruff/issues/6157
- https://github.com/astral-sh/ruff/issues/8157

Is it possible to have a rule that bans something like this:

```py
def my_function():
    # ruff: noqa: PLR0914
    ...
```

This is a valid noqa which ruff currently respects, but usually isn't what the author intended; they intended to ignore the rule for the function but not the whole file.  Something like "only allow `ruff: noqa: ` near the top of the file" might make sense, but disallowing `ruff: noqa: ` comments that aren't at the start of a line (including whitespace) seems maybe better defined?
